include/triton/Analysis/Allocation.h

```diff
@@ -102,7 +108,8 @@ class Allocation {
   explicit Allocation(Operation *operation) : operation(operation) {}
 
   /// Runs allocation analysis on the given top-level operation.
-  void run(FuncAllocMapT &funcAllocMap);
+  void run(FuncAllocMapT &funcAllocMap,
+           triton::AllocationAnalysisScratchSizeFn scratchSizeGetter);
```

Comment:
Is it possible to set `scratchSizeGetter` in the constructor?

Reply:
Sure, but wouldn't that lead to extra space allocated for the function pointer when we only need it in the `run` call?

Reply:
IMO that's not an issue

Reply:
We have several `Allocation` instances and this would only add unneeded overhead. I'd rather keep as is

Reply:
Bear in mind we'd be storing collections of `Allocation` instances with the same `scratchSizeGetter` as a member.

Reply:
Alright, I'm OK with it